.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,50 @@
+name: Shoppy E2E tests
+
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    branches:
+      - "main"
+
+jobs:
+  e2e-tests:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    name: e2e-tests
+    env:
+      PREMIUM_EMBEDDING_TOKEN: ${{ secrets.ENTERPRISE_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Run Shoppy in Docker
+        run: |
+          cp .env.docker.example .env.docker &&
+          yarn docker:up -d &&
+          while ! nc -z localhost 4400; do sleep 1; done
```

Comment:
Since we already have `timeout-minutes: 10` it should eventually stop the run if `localhost:4400` is not up for some reason 👍🏻

Reply:
We need it for a different purpose:

- we run docker in a detached mode (-d flag), so we have to wait till the client container is healthy before we run the next step